hpx/plugins/parcelport/mpi/mpi_environment.hpp

```diff
@@ -29,27 +30,38 @@ namespace hpx { namespace util
         static int rank();
         static int size();
 
+        static MPI_Comm& communicator();
+
         static std::string get_processor_name();
 
+        struct scoped_lock
+        {
+            scoped_lock();
+            ~scoped_lock();
+        };
+
+        struct scoped_try_lock
+        {
+            scoped_try_lock();
+            ~scoped_try_lock();
+            bool locked;
+        };
+
+        typedef hpx::lcos::local::spinlock mutex_type;
```

Comment:
The non atomic is bool is fine. There should only be a single reference to a object of scoped_lock anyways. Probably should make that non copyable and non movable. (compare to boost::unique_lock).